From: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
Date: Mon, 15 Jan 2024 21:42:59 +0100
Subject: [PATCH 1/1] plugins: FromHost must be 63 characters long

According to RFC 1035 the hostname without domain may be up to 63
characters long. Consider this when creating SQL databases.

Signed-off-by: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
Forwarded: https://github.com/rsyslog/rsyslog/pull/5310
Last-update 2024-01-15
---
 plugins/ommysql/createDB.sql | 2 +-
 plugins/ompgsql/createDB.sql | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/plugins/ommysql/createDB.sql b/plugins/ommysql/createDB.sql
index 211cfb0e2..626128465 100644
--- a/plugins/ommysql/createDB.sql
+++ b/plugins/ommysql/createDB.sql
@@ -8,7 +8,7 @@ CREATE TABLE SystemEvents
         DeviceReportedTime datetime NULL,
         Facility smallint NULL,
         Priority smallint NULL,
-        FromHost varchar(60) NULL,
+        FromHost varchar(63) NULL,
         Message text,
         NTSeverity int NULL,
         Importance int NULL,
diff --git a/plugins/ompgsql/createDB.sql b/plugins/ompgsql/createDB.sql
index 0c0f7e13a..2f1ade046 100644
--- a/plugins/ompgsql/createDB.sql
+++ b/plugins/ompgsql/createDB.sql
@@ -8,7 +8,7 @@ CREATE TABLE SystemEvents
         DeviceReportedTime timestamp without time zone NULL,
         Facility smallint NULL,
         Priority smallint NULL,
-        FromHost varchar(60) NULL,
+        FromHost varchar(63) NULL,
         Message text,
         NTSeverity int NULL,
         Importance int NULL,
-- 
2.43.0

